lorien/InfiniteCanvas/InfiniteCanvas.gd

```diff
@@ -46,21 +46,23 @@ func _ready():
 	_viewport.size = OS.window_size
 
 # -------------------------------------------------------------------------------------------------
-func _input(event: InputEvent) -> void:
+func _gui_input(event: InputEvent) -> void:
 	if event is InputEventMouseMotion:
 		info.current_pressure = event.pressure
 
-# -------------------------------------------------------------------------------------------------
-func _process(delta: float) -> void:
-	# Deselect selected strokes with shortcut key
-	if Input.is_action_just_pressed("deselect_all_strokes"):
+	if event.is_action("deselect_all_strokes"):
 		if _active_tool == _selection_tool:
 			_selection_tool.deselect_all_strokes()
-	
-	# Delete selected strokes with shortcut key
-	if Input.is_action_just_pressed("delete_selected_strokes"):
+
+	if event.is_action("delete_selected_strokes"):
 		if _active_tool == _selection_tool:
 			_delete_selected_strokes()
+	
+	if ! get_tree().is_input_handled():
+		_camera.tool_event(event)
+	if ! get_tree().is_input_handled():
+		if _active_tool.enabled:
+			_active_tool.tool_event(event)
```

Comment:
This is an interesting different change: To prevent tools from having to rely on the `_input` function, I gave them a dedicated callback that is explicitly called by InfiniteCanvas. As can be deduced from the if-statement, these will respect the "input handled" flag to prevent double processing in case someone defines overlapping keybindings, which can happen until all keybindings are rebindable.

How can this happen? For example the `player_` events and `ui_` events are not rebindable, but might generate overlap with other actions.